### PR TITLE
MueLu: Take two on aggregate quality

### PR DESCRIPTION
**Pull request by csiefer2**
Dirty work by @dsunder.
Tests: RHEL6

### CODE REVIEW

**Comment by csiefer2** [comment]
@trilinos/framework cdash links aren't working.  Help?

File: packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp [hunk 1]
```diff
@@ -358,27 +396,69 @@ namespace MueLu {
...
-
     }
 
+
+    if (num_bad_aggs > 0) mean_bad_agg /= num_bad_aggs;
+    mean_good_agg /= agg_qualities->getLocalLength() - num_bad_aggs;
+
     if (num_bad_aggs == 0) {
-      GetOStream(Statistics1) << "All aggregates passed the quality measure. Worst aggregate had quality " << worst_agg << std::endl;
+      GetOStream(Statistics1) << "All aggregates passed the quality measure. Worst aggregate had quality " << worst_agg << ". Mean aggregate quality " << mean_good_agg << "." << std::endl;
```
Comment (jhux2) [inline packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp]: Same comment as before.  I don't think that a process `i !=root` will print, even if `num_bad_aggs>0` on proc `i`.

Comment (csiefer2) [reply packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp]: Yup.

I don't think we have a way to control for output level w/o controlling processor count.

Comment (cgcgcg) [reply packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp]: Global reduce over `num_bad_aggs`?

Comment (jhux2) [reply packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp]: @cgcgcg That was what I was thinking.  O/W there's failure w/o a message.  (There's a joke in there somewhere.)

**Pull request merged**